routes/uploads.js

```diff
@@ -144,6 +158,105 @@ module.exports = [
   },
 
   /**
+   * @api {post} /uploads/:id/:sceneIdx/:imageId Update imagery metadata
```

Comment:
I don't fully understand the purpose of this endpoint. I think it's run after `process.sh` has completed (whether sucessfully or not)? But I don't understand why it's over HTTP? `process.sh` has direct access to the DB, so why not just do this in `process.sh`? Or even better, in fact ideally, in `bin/transcoder.js`?

Reply:
It's used for marking status generally--requests will be made when transcoding starts, when different stages are reach, and when it completes (successfully or not).

I'm trying to treat `process.sh` as a black-boxed external dependency to the extent I'm able. Mapzen uses this same script to transcode DEMs and needs to write footprint data to Postgres, so HTTP seemed like the simplest way to decouple them.

When I add the optional Batch transcoding backend (as an alternative to the `bin/transcoder.js`monq worker), this may make more sense (transcoding jobs are fully queued rather than having "callbacks" associated ith them--it doesn't seem to make sense to have each transcoding node open a connection to Mongo.

Reply:
> it doesn't seem to make sense to have each transcoding node open a connection to Mongo
Can you explain this in detail please?

I think there are a few problems with this approach. As mentioned, using HTTP, when there's a perfectly good Mongo socket connection already there. Supporting Batch is out of scope for this work, not to mention that Batch is platform-specific - in fact I'm sure that the "on-demand worker instance" paradigm is significantly cheaper almost everywhere apart from AWS. If we're going to be accomodating said paradigm, then there's broader benefits to supporting a platform agnostic method. However, I don't think OAM's transcoding load is going to scale to the point of seeing architectural cost-benefits for a long time and even then, the simpler approach is just to host somewhere cheaper. Also, `process.sh` should first and foremost be serving OAM, it doesn't need to cater to anyone else. And even if theoretically there was a justification for HTTP here, then strictly speaking it should exist as a seperate API, therefore a distinct service. Having this endpoint on the catalog is unintuitive and sets a bad precedent of blurring seperation of concerns.

Reply:
@smit1678 is Batch out of scope? I've been assuming that it's not, for these reasons:

* we'd planned on incorporating it when the uploader got refactored to transcode imagery, but Nick ran out of time
* OAM occasionally needs to handle ~50GB uploads and process them; in the current paradigm, this means keeping hundreds of GB of space available for when this does happen
* after disasters, large quantities of imagery become available and should be ingested into OAM (I just did this with DG's hurricane-related imagery this/last week). being able to do this rapidly benefits many parties. using Batch makes this process much quicker--Irma imagery took ~12 hours to fully ingest because it could only process 4 images at a time (and I managed to use up the burst credits on the t2 instance; granted, an AWS-specific thing).
* it represents progressive enhancement when running on AWS; the existing worker implementation (mongodb/monq) can be used when Batch isn't available

marblecutter-tools (which includes `process.sh`) was extracted and made generic based on the transcoding utilities in oam-dynamic-tiler. Mapzen *is* using these for DEM data, and re-adding application-specific code will only make it diverge again.

Transcoding is logically a separate service ("service / tool that produces COGs") distinct from the catalog API (which deals in metadata management + uploads). Without adding application-specific service dependencies (mongo), do you have any suggestions on how to communicate progress, status, and eventually submit generated metadata about items that have been queued?

Reply:
> is Batch out of scope? I've been assuming that it's not, for these reasons:

Correct, I don't think this is out of scope at all and is something we've been thinking about for a while. But I don't think scope is the right question here. Let's agree upon implementation method for how we talked about extracting and refactoring the dynamic-tiler into a set of tools. Let's make sure we're on the same page there.